core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/function/NumberSelection.java

```diff
@@ -0,0 +1,222 @@
+
+          @Override
+          public void beginFunction() {
+            startInt();
+          }
+
+          @Override
+          public void operateOn(StructuredRecord record) {
+            operateOnInt(record);
+          }
+
+          @Override
+          public void finishFunction() {
+            finishInt();
+          }
+        };
+        break;
+      case LONG:
+        delegate = new SelectionFunction() {
+          @Override
+          public List<StructuredRecord> getSelectedRecords() {
+            return getRecords();
+          }
+
+          @Override
+          public void beginFunction() {
+            startLong();
+          }
+
+          @Override
+          public void operateOn(StructuredRecord record) {
+            operateOnLong(record);
+          }
+
+          @Override
+          public void finishFunction() {
+            finishLong();
+          }
+        };
+        break;
+      case FLOAT:
+        delegate = new SelectionFunction() {
+          @Override
+          public List<StructuredRecord> getSelectedRecords() {
+            return getRecords();
+          }
+
+          @Override
+          public void beginFunction() {
+            startFloat();
+          }
+
+          @Override
+          public void operateOn(StructuredRecord record) {
+            operateOnFloat(record);
+          }
+
+          @Override
+          public void finishFunction() {
+            finishFloat();
+          }
+        };
+        break;
+      case DOUBLE:
+        delegate = new SelectionFunction() {
+          @Override
+          public List<StructuredRecord> getSelectedRecords() {
+            return getRecords();
+          }
+
+          @Override
+          public void beginFunction() {
+            startDouble();
+          }
+
+          @Override
+          public void operateOn(StructuredRecord record) {
+            operateOnDouble(record);
+          }
+
+          @Override
+          public void finishFunction() {
+            finishDouble();
+          }
+        };
+        break;
+      default:
+        throw new IllegalArgumentException(String.format("Field '%s' is of unsupported non-numeric type '%s'. ",
+                                                         fieldName, fieldType));
+    }
+  }
+
+  @Override
+  public void beginFunction() {
+    delegate.beginFunction();
+  }
+
+  @Override
+  public void operateOn(StructuredRecord record) {
+    delegate.operateOn(record);
+  }
+
+  @Override
+  public void finishFunction() {
+    delegate.finishFunction();
+  }
+
+  @Override
+  public List<StructuredRecord> getSelectedRecords() {
+    return delegate.getSelectedRecords();
+  }
+
+  public String getFieldName() {
+    return fieldName;
+  }
+
+  protected abstract void startInt();
+
+  protected abstract void startLong();
+
+  protected abstract void startFloat();
+
+  protected abstract void startDouble();
+
+  protected abstract void operateOnInt(StructuredRecord record);
```

Comment:
would be easier for subclasses if this method got both the int value and the record as arguments, since the first thing they will always do is look up the int value anyway.